engine/src/exec-strings.cpp

```diff
@@ -1993,26 +1993,15 @@ void MCStringsExecFilterRegex(MCExecContext& ctxt, MCStringRef p_source, MCStrin
     delete matcher;
 }
 
-void MCStringsExecFilterWildcardIntoIt(MCExecContext& ctxt, MCStringRef p_source, MCStringRef p_pattern, bool p_without, bool p_lines)
+void MCStringsExecFilterExpression(MCExecContext& ctxt, MCStringRef p_source, MCExpression* p_expression, bool p_without, bool p_lines, MCStringRef &r_result)
 {
-    MCAutoStringRef t_result;
-    MCStringsExecFilterWildcard(ctxt, p_source, p_pattern, p_without, p_lines, &t_result);
+    // Create the pattern matcher
+    MCPatternMatcher *matcher;
+    matcher = new (nothrow) MCExpressionMatcher(p_expression, p_source, ctxt . GetStringComparisonType());
```

Comment:
There doesn't seem to be any need for this to be heap-allocated...
```
MCPatternMatcher matcher(p_expression, ...)
```
Should have the same effect (just pass as `&matcher`)

Reply:
The same is true for the FilterRegex and FilterWildcard cases - probably worth adding the same change to them (but as a separate commit for clarity).